### PR TITLE
[Web Portal] fix port list bug

### PR DESCRIPTION
**Pull request by sunqinzheng**

### CODE REVIEW

File: src/webportal/src/app/job-submission/models/protocol-schema.js [hunk 1]
```diff
@@ -44,12 +44,12 @@ export const taskRoleSchema = Joi.object().keys({
     cpu: Joi.number().required(),
     memoryMB: Joi.number().required(),
     gpu: Joi.number().required(),
-    ports: Joi.object().pattern(/^[a-zA-Z_][a-zA-Z0-9_]*$/, Joi.number()),
+    ports: Joi.object().pattern(/^[A-Za-z0-9\-._~]+$/, Joi.number()),
```
[the remark this answers is not in the document]
Comment (sunqinzheng) [reply src/webportal/src/app/job-submission/models/protocol-schema.js]: The pattern in protocol's [schema](https://github.com/microsoft/pai/blob/0f1d8d2fe4d67694d5d5bb95aaabe59be07928c4/docs/pai-job-protocol.yaml#L67) is different

[the remark this answers is not in the document]
Comment (sunqinzheng) [reply src/webportal/src/app/job-submission/models/protocol-schema.js]: I can fix this but I think we need to create a seperate issue to track the inconsistent document problem.

[the remark this answers is not in the document]
Comment (Binyang2014) [reply src/webportal/src/app/job-submission/models/protocol-schema.js]: Got it, will ask @abuccts to check

[the remark this answers is not in the document]
Comment (Binyang2014) [reply src/webportal/src/app/job-submission/models/protocol-schema.js]: Checked with yifan, better to fix this and create another issue for inconsistent document

[the remark this answers is not in the document]
Comment (sunqinzheng) [reply src/webportal/src/app/job-submission/models/protocol-schema.js]: #3243

**Pull request merged**